src/change_stream.ts

```diff
@@ -183,6 +183,14 @@ export interface ChangeStreamDocumentCommon {
    */
   clusterTime?: Timestamp;
 
+  /**
+   * The server date and time of the database operation.
+   * wallTime differs from clusterTime in that clusterTime is a timestamp taken from the oplog entry associated with the database operation event.
+   * The format is "YYYY-MM-DD HH:MM.SS.millis".
+   * @sinceServerVersion 6.0.0
+   */
+  wallTime?: Date;
```

Comment:
According to the docs, `wallTime` isn't present for [reshardCollection events](https://www.mongodb.com/docs/manual/reference/change-events/reshardCollection/#mongodb-data-reshardCollection) and [refineCollectionShardKey](https://www.mongodb.com/docs/manual/reference/change-events/refineCollectionShardKey/#mongodb-data-refineCollectionShardKey) events.  That means we probably shouldn't include `wallTime` on `ChangeStreamDocumentCommon`.  

Can we update the implementation to only include wallTime where it actually can be present?  We might also want tests asserting that it is present on all types we expect it to be on.

Also - Can we clean up the TSDoc comment?  If it is of type Date, the string format section doesn't really make much sense.

Reply:
Good catch. I've updated it to only be included where it is present and updated the comment.